linked-list.rb

```diff
@@ -55,20 +55,44 @@ def display
        current = @head
        full_list = []
        while current.next_node != nil
-           full_list += [current.value.to_s]
+           full_list << current.value.to_s
            current = current.next_node
        end
        full_list += [current.value.to_s]
        puts full_list.join("->")
    end
 
+# doing while current.next_node != nil prevent code from ever looking
+# at the last node, so they wouldn't get evaluated using the pattern above
    def include?(key)
+    current = @head
+    while current != nil 
+      return true if current.value == key
+      current = current.next_node
+    end
+    return false
    end
 
    def size
+    @size = 0
+    current = @head
+    while current != nil
+      @size += 1
+      current = current.next_node
+    end
+    return @size
    end
 
    def max
+    current = @head
+    maximum = current.value
+    while current != nil
+      if current.value > maximum
+        maximum = current.value
+      end
+      current = current.next_node
+    end
+    return maximum
    end
```

Comment:
I LOVE that you recognized that you could change your loop to `while current != nil` instead of `while current.next_node != nil` ! I told you to model after display which does it the latter way -- because sometimes you need to do something different with the last case and I wanted that to be the pattern I showed -- but now I want to point out how we could have optimized the pattern just slightly. Really nice!